curiosrc/winning/winning_task.go

```diff
@@ -433,6 +435,47 @@ func (t *WinPostTask) Do(taskID harmonytask.TaskID, stillOwned func() bool) (don
 	return true, nil
 }
 
+func (t *WinPostTask) generateWinningPost(
+	ctx context.Context,
+	ppt abi.RegisteredPoStProof,
+	mid abi.ActorID,
+	sectors []storiface.PostSectorChallenge,
+	randomness abi.PoStRandomness) ([]proof.PoStProof, error) {
+
+	// don't throttle winningPoSt
+	// * Always want it done asap
+	// * It's usually just one sector
+	var wg sync.WaitGroup
+	wg.Add(len(sectors))
+
+	vproofs := make([][]byte, len(sectors))
+	var rerr error
+
+	for i, s := range sectors {
+		go func(i int, s storiface.PostSectorChallenge) {
+			defer wg.Done()
+
+			vanilla, err := t.paths.GenerateSingleVanillaProof(ctx, mid, s, ppt)
+			if err != nil {
+				rerr = multierror.Append(rerr, xerrors.Errorf("get winning sector:%d,vanila failed: %w", s.SectorNumber, err))
```

Comment:
This is technically a racy assignment; Realistically not a problem because most (all?) of the time we just prove one sector and usually don't fail here; Still would be more correct to wrap this in some lock

Reply:
mind that I got rid of the multi-err since it's typically only 1 sector which can only make 1 err?